File: ignored-test/csharp-xunit/CustomArgumentException_RecursiveAsyncExceptionWithoutFlattenException.cs
using Xunit;

namespace None
{
    public class CustomArgumentException
    {
        [Fact(Skip = "")]
        public void RecursiveAsyncExceptionWithoutFlattenException()
        {
            {
                var recursionCount = 3;
                Func<int> innerAction = () => throw new ApplicationException("Life is hard");
                var t1 = System.Threading.Tasks.Task<int>.Factory.StartNew(() =>
                {
                    return NestedFunc(recursionCount, innerAction);
                });
                try
                {
                    t1.Wait();
                }
                catch (AggregateException ex)
                {
                    var layoutRenderer = new ExceptionLayoutRenderer()
                    {
                        Format = "ToString",
                        FlattenException = false
                    };
                    var logEvent = LogEventInfo.Create(LogLevel.Error, null, null, (object)ex);
                    var result = layoutRenderer.Render(logEvent);
                    int needleCount = 0;
                    int foundIndex = result.IndexOf(nameof(NestedFunc), 0);
                    while (foundIndex >= 0)
                    {
                        ++needleCount;
                        foundIndex = result.IndexOf(nameof(NestedFunc), foundIndex + nameof(NestedFunc).Length);
                    }
            
                    Assert.True(needleCount >= recursionCount, $"{needleCount} too small");
                }
            }
        }
    }
}

File: ignored-test/python-pytest/CustomArgumentException_RecursiveAsyncExceptionWithoutFlattenException.py
import pytest

class TestCustomArgumentException:

    @pytest.mark.skip(reason="")
    def test_recursive_async_exception_without_flatten_exception(self):
        recursion_count = 3
        def inner_action():
            raise ApplicationException("Life is hard")
        t1 = System.Threading.Tasks.Task[int].factory.start_new(
            lambda: NestedFunc(recursion_count, inner_action)
        )
        try:
            t1.wait()
        except AggregateException as ex:
            layout_renderer = ExceptionLayoutRenderer()
            layout_renderer.format = "ToString"
            layout_renderer.flatten_exception = False
            log_event = LogEventInfo.create(LogLevel.error, None, None, ex)
            result = layout_renderer.render(log_event)
            needle_count = 0
            found_index = result.find("NestedFunc", 0)
            while found_index >= 0:
                needle_count += 1
                found_index = result.find("NestedFunc", found_index + len("NestedFunc"))
            assert needle_count >= recursion_count, f"{needle_count} too small"
